### DISCUSSION TITLE
Running async method OnCofirm (in <Popconfirm> component)

### DISCUSSION THREAD

**Discussion by bobtnt** [ant-design-blazor/ant-design-blazor#2791] [Q&A] [answered]
I have an async method that I want to run OnConfirm. The async method contains a line of code that I await and another line of code that I don't await.

With my own button,  **@onclick="async() => await myAsyncMethod"**  will do what I want, i.e. run the awaited line and wait for it to complete, and then exit myAsyncMethod without waiting for the second line of code that I don't await.

With Popconfirm, when I do **OnConfirm="async() => await myAsyncMethod"** it will await everything in myAsyncMethod (both first and second line of code). How do I fix this?

Thank you very much!

**Comment by ElderJames** [comment, marked as answer]
Hello @bobtnt , it work well for me with following code. I don't know if that's what you mean. If not, please provide your demo.

```
@inject IMessageService _message

<Popconfirm Title="Are you sure delete this task?"
            OnConfirm="Confirm"
            OkText="OK">
    <a>Delete</a>
</Popconfirm>

@code{

    private async Task Confirm()
    {
        await _message.Success("Clicked on Yes");
        
        _message.Success("Clicked on Yes 2",10);
    }
}
```

**Reply by bobtnt** [reply]
Thank you very very much for such a quick response! You're right, there's no issue at all with PopConfirm. 

The issues were due to my lack of knowledge. Some notes in case it's useful for others in the future:
1. Using Thread.Sleep causes UI to freeze. Use Task.Delay instead
2. No need to do OnConfirm="async () => await Confirm()" . Just do OnConfirm="Confirm" as shown in James' example above
3. Use "async Task" instead of "async void".
